myFslInstallerScript.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+#myfslinstaller.sh
+
+set -e
+set -u
+set -o nounset
+
+#myfslinstaller function takes the parameters (OS and Version) and downloads the corresponding file of fsl installer from the server.
+if [[ $# -eq 0 ]] ; then
+    echo 'No parameters are passed, taking default values'
+fi
+
+version=${1:-5.0.6}
+os=${2:-CentOS7}
+installDir=${3:-/usr/local/src}
+downloadDir=${4:-/usr/local/etc/}
+
+function myFslInstaller {
+echo "###----FSL INSTALLER---###"
+echo "Version of FSL:$version"
+echo "OS:$os"
+echo "Installation Directory:$installDir"
+echo "Download Directory: $downloadDir"
+
+if [[ ! ("$version" =~ ^[0-9].+$) ]]; then 
+    echo 'This does not look like a valid version number'
+    exit 1
+fi
+
+url=$(getDownloadURL "$version" "$os")
+
+downloadFileName=$(echo "$url"|sed 's#.*/##')
+echo "$downloadFileName"
+
+fslDownloadAbsoluteFileName=$downloadDir"$downloadFileName"
+echo "$fslDownloadAbsoluteFileName"
+
+valid=$(validURL "$url")
+echo "Valid: $valid"
+if [[ "$valid" == true ]]; then
+   mkdir -p "$downloadDir"
+     if [ ! -f "$fslDownloadAbsoluteFileName" ]; then
+    	download "$url" "$fslDownloadAbsoluteFileName"
+     fi
+else
+ echo "File doesn't exist in server"
+ exit 1
+fi
+
+md5Url=$(getMD5Url "$version" "$os")
+echo "$md5Url"
+
+md5DownloadAbsoluteFileName=$downloadDir"$version"".txt"
+
+download "$md5Url" "$md5DownloadAbsoluteFileName"
+
+isMD5Equal=$(validateMD5 "$md5DownloadAbsoluteFileName" "$fslDownloadAbsoluteFileName")
+
+if [[ "$isMD5Equal" ]];then
+	untar "$fslDownloadAbsoluteFileName"
+else
+	echo "MD5 checksum does not match"
+fi
+
+}
+
+function getMD5Url {
+version="$1"
+os="$2"
+os="${os,,}"
+main_url='http://fsl.fmrib.ox.ac.uk/fsldownloads/md5sums/'
+if [[ "$os" == "centos7" ]]; then
+	echo "$main_url""fsl-""$version""-""centos6_64.tar.gz.md5"
```

Comment:
FSL for CentOS 6 is compatible with CentOS 7 as per documentation. The validity check of url will throw error in case we are removing the check for version of the OS. I have added a warning now.